### PR TITLE
Compat for chol(::UniformScaling) and fallback for logdet()

### PR DESCRIPTION
**Pull request by mschauer**
As suggested in https://github.com/JuliaLang/METADATA.jl/pull/10316. I did not hook into ``_chol!`` because that interface is private and volatile.

### CODE REVIEW

**Comment by tkelman** [comment]
Thanks, could you add a note to the readme about these features?

File: src/Compat.jl [hunk 1]
```diff
@@ -607,6 +607,19 @@ if VERSION < v"0.6.0-pre.beta.455"
     isless(x::Dates.OtherPeriod, y::Dates.FixedPeriod) = throw(MethodError(isless, (x, y)))
 end
 
+# https://github.com/JuliaLang/julia/pull/22629
+if VERSION < v"0.7.0-DEV.773"
```
Comment (tkelman) [inline src/Compat.jl]: I'm getting
```
$ contrib/commit-name.sh ad9f5defaefd4b1ef75db928b13c8492bb9b734e
0.7.0-DEV.848
```

Comment (mschauer) [inline src/Compat.jl]: Ah, I see, my own commit was 
```
0f6d354109c2f039c2aa04dc646f3cc89c5c37f9
0.7.0-DEV.773
```
but the commit number when merged is 
```
ad9f5defaefd4b1ef75db928b13c8492bb9b734e
0.7.0-DEV.848
```
Maybe a comment in the README is in order.

File: src/Compat.jl [hunk 2]
```diff
@@ -607,6 +607,19 @@ if VERSION < v"0.6.0-pre.beta.455"
...
+    import Base: logdet
+    logdet(A) = log(det(A))
+end
+
+# https://github.com/JuliaLang/julia/pull/22633
+if VERSION < v"0.7.0-DEV.1041"
+    import Base.LinAlg: chol, chol!
```
Comment (mschauer) [inline src/Compat.jl]: Here the commit number did not change, " andreasnoack merged commit eb133ab into JuliaLang:master"
```
contrib/commit-name.sh eb133ab
0.7.0-DEV.1041
```

File: src/Compat.jl (hunk 1 above)
Comment (tkelman) [inline src/Compat.jl]: yeah for some reason I thought there was already a note to always use the merge commit, but guess not yet

**Comment by mschauer** [comment]
Squashed and rebased.

**Pull request merged**